megamek/src/megamek/server/victory/EnemyCmdrDestroyedVictory.java

```diff
@@ -22,60 +28,45 @@
 import java.util.Map;
 
 /**
- * implements "enemy commander destroyed"
+ * Implements a victory condition that checks if all enemy commanders have been killed.
  */
-public class EnemyCmdrDestroyedVictory implements IVictoryConditions, Serializable {
-    private static final long serialVersionUID = 2525190210964235691L;
-
-    public EnemyCmdrDestroyedVictory() {
-
-    }
+public class EnemyCmdrDestroyedVictory implements VictoryCondition, Serializable {
 
     @Override
-    public VictoryResult victory(Game game, Map<String, Object> ctx) {
-        VictoryResult vr = new VictoryResult(true);
+    public VictoryResult checkVictory(Game game, Map<String, Object> ctx) {
+        VictoryResult victoryResult = new VictoryResult(true);
         // check all players/teams for killing enemy commanders
         // score is 1.0 when enemy commanders are dead
-        boolean victory = false;
+        boolean isVictory = false;
         HashSet<Integer> doneTeams = new HashSet<>();
-        for (Player player : game.getPlayersVector()) {
-            boolean killedAll = true;
+
+        for (Player player : game.getPlayersList()) {
             int team = player.getTeam();
             if (team != Player.TEAM_NONE) {
                 if (doneTeams.contains(team)) {
+                    // only consider each team once
                     continue;
                 }
-                // skip if already dealt with this team
                 doneTeams.add(team);
             }
-            for (Player enemyPlayer : game.getPlayersVector()) {
-                if (enemyPlayer.equals(player) ||
-                        (team != Player.TEAM_NONE && team == enemyPlayer.getTeam())) {
-                    continue;
-                }
-                if (game.getLiveCommandersOwnedBy(enemyPlayer) > 0) {
-                    killedAll = false;
-                }
-            }
-            // all enemy commanders are dead
-            if (killedAll) {
+
+            boolean killedAllCommanders = game.getPlayersList().stream()
+                    .filter(p -> p.isEnemyOf(player))
+                    .mapToInt(game::getLiveCommandersOwnedBy).sum() == 0;
+
+            if (killedAllCommanders) {
+                Report r = new Report(7110, Report.PUBLIC);
                 if (team == Player.TEAM_NONE) {
-                    Report r = new Report(7110, Report.PUBLIC);
                     r.add(player.getName());
-                    vr.addPlayerScore(player.getId(), 1);
-                    vr.addReport(r);
+                    victoryResult.setPlayerScore(player.getId(), 1);
```

Comment:
The former addPlayerScore() always "set" (it used .put() in the score map). That's why I renamed it. The scores of different victories are added in VictoryHelper, l. 105 (`combinedResult.addScores(victoryResult);`) I called this method addScores because it really adds.